tools/clang/tools/dxcompiler/dxcutil.cpp

```diff
@@ -52,19 +52,22 @@ namespace {
 bool CreateValidator(CComPtr<IDxcValidator> &pValidator,
                      hlsl::options::ValidatorSelection SelectValidator =
                          hlsl::options::ValidatorSelection::Auto) {
+  bool bInternalValidator = false;
   bool bInternal =
       SelectValidator == hlsl::options::ValidatorSelection::Internal;
   bool bExternal =
       SelectValidator == hlsl::options::ValidatorSelection::External;
-  if (!bInternal && DxilLibIsEnabled())
-    DxilLibCreateInstance(CLSID_DxcValidator, &pValidator);
 
-  bool bInternalValidator = false;
-  if (pValidator == nullptr) {
-    IFTBOOL(!bExternal, DXC_E_VALIDATOR_MISSING);
+  if (!bExternal) {
     IFT(CreateDxcValidator(IID_PPV_ARGS(&pValidator)));
     bInternalValidator = true;
   }
+
+  if (pValidator == nullptr && DxilLibIsEnabled()) {
```

Comment:
hmm...it looks to me that this function could now exit, without throwing, with pValidator set to nullptr.  Previously this wasn't possible.

I think both the original, and this version, are pretty hard to follow.  Some comments, or maybe even refactoring, could help here.

Reply:
The new version is a lot clearer!